src/scripts/main.js

```diff
@@ -1,3 +1,28 @@
 'use strict';
 
-// write code here
+const tree = document.querySelector('.tree');
+const listItems = document.querySelectorAll('li');
+
+listItems.forEach(item => {
+  if (item.firstElementChild) {
+    const span = document.createElement('span');
+
+    span.textContent = item.childNodes[0].textContent.trim();
+    item.childNodes[0].textContent = '';
+    item.prepend(span);
+  }
+});
+
+tree.addEventListener('click', e => {
+  if (e.target.tagName === 'SPAN') {
+    const children = e.target.parentNode.lastElementChild.children;
+
+    for (const child of children) {
+      if (!child.style.display) {
```

Comment:
It's better to determine visibility by adding and removing CSS classes, not by directly manipulating the style properties of the element.